### PR TITLE
[fix](nereids)filter estimation for slot=unknown

### PR DESCRIPTION
**Pull request by englefly**
## Proposed changes
detect new pattern: slot=unknown
suppose slot.ndv = 5, slot.row=100
expect filter result row is 20, but in master, the result is 0

Issue Number: close #xxx

### CODE REVIEW

**Comment by englefly** [comment]
run buidlall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/stats/FilterEstimation.java [hunk 1]
```diff
@@ -331,14 +331,28 @@ private Statistics estimateEqualTo(ComparisonPredicate cp, ColumnStatistic stats
...
-        double val = statsForRight.maxValue;
-        if (val > statsForLeft.maxValue || val < statsForLeft.minValue) {
-            selectivity = 0.0;
+        if (statsForLeft.isUnKnown) {
+            selectivity = DEFAULT_INEQUALITY_COEFFICIENT;
         } else {
-            selectivity = StatsMathUtil.minNonNaN(1.0, 1.0 / ndv);
+            double ndv = statsForLeft.ndv;
+            if (statsForRight.isUnKnown) {
+                if (ndv >= 1.0) {
```
Comment (xzj7019) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/stats/FilterEstimation.java:339]: what is the case where statsForLeft is not unknown but ndv < 1.0? Should we do a normalization if the ndv < 1.0?

Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/stats/FilterEstimation.java:339]: this ndv is from left, and right is unknown

**Review by starocean999 — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/stats/FilterEstimation.java (hunk 1 above)
Comment (xzj7019) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/stats/FilterEstimation.java:339]: I mean whether we should do a data normalization for those concept which has certain physical properties, such as row count, ndv, selectivity, etc. We may do this at next generation code model.

**Review by xzj7019 — APPROVED** [review]

**Pull request merged**